src/plugins/default/anchors.js

```diff
@@ -0,0 +1,40 @@
+const cheerio = module.parent.require('cheerio');
+const md = require('./../../lib/markbind/src/lib/markdown-it');
+
+const ANCHOR_HTML = '<a class="fa fa-anchor" href="#"></a>';
+
+/**
+ * Generates a heading selector based on the indexing level
+ * @param headingIndexingLevel to generate
+ */
+function generateHeadingSelector(headingIndexingLevel) {
+  let headingsSelector = 'h1';
+  for (let i = 2; i <= headingIndexingLevel; i += 1) {
+    headingsSelector += `, h${i}`;
+  }
+  return headingsSelector;
+}
+
+/**
+ * Adds anchor links to headers
+ */
+module.exports = {
+  postRender: (content, pluginContext, frontMatter, page) => {
```

Comment:
I needed access to `headingIndexingLevel,` so I added a 4th parameter, page which is the `Page` object itself... Other plugins might need access to page config so this could be helpful

I'm thinking we can use this for internal plugins, but don't document it, since we don't want to expose authors to the implementation

Reply:
I'm just thinking it might be dangerous to pass the entire `Page` object down to plugins, since that would allow the plugin to modify or invoke the variables and methods of `Page` which could be dangerous. 😅 

What do you think about passing down a copy of the `pageConfig` object instead? Or maybe having a helper method `getDefaultPluginsContext` which constructs the context for default plugins and appends it to the other `pluginsContext`?

Reply:
Sure, I think I'll do the page config method, I think `getDefaultPluginsContext` might end up growing pretty large if we have a lot of default plugins =P

Reply:
This part doesn't seem to be addressed?